torch/optim/optimizer.py

```diff
@@ -34,6 +35,18 @@ def __init__(self, params, defaults):
         torch._C._log_api_usage_once("python.optimizer")
         self.defaults = defaults
 
+        def _step_with_profile(func, class_name):
+            """Wrap self.step under profiler. In order to avoid of instrumenting each sub-class."""
+            @functools.wraps(func)
+            def wrapper(*args, **kwargs):
+                name = str.format("{}#{}.{}", "Optimizer.step", class_name, "step")
```

Comment:
can you pull it out so it's initialized once and not on every call?

also `"{}".format(...)` is more pythonic

Reply:
Thanks. Fixed.